src/map/atcommand.c

```diff
@@ -6727,23 +6727,27 @@ ACMD(refreshall)
 }
 
 /*==========================================
- * @identify
+ * @identify / @identifyall
  * => GM's magnifier.
  *------------------------------------------*/
 ACMD(identify)
 {
 	int num = 0;
+	bool identifyall = (strcmpi(info->command, "identifyall") == 0);
 
 	for (int i = 0; i < sd->status.inventorySize; i++) {
-		if(sd->status.inventory[i].nameid > 0 && sd->status.inventory[i].identify!=1){
+		if (sd->status.inventory[i].nameid > 0 && sd->status.inventory[i].identify != 1) {
+			if (identifyall)
```

Comment:
perhaps  move this line to before the `for-loop`? so that it won't need to go through the loop everytime even if we are just using `@identify` to identify one item.

Reply:
I was wondering that too, but it would duplicate most of the loop code so I wasn't sure if I should

Reply:
@Emistry it needs to loop even if you use `@identify`, to know how many item needs to be identified.

Reply:
Yes, it is as dastgir said. Can I mark as resolved and leave it as is?

Reply:
for faster code ``if (identifyall)`` better move outside of ``for``
this mean for faster code need two ``for`` one inside if block and second inside else block

Reply:
Done. I'm wondering, would make more sense to make a separate command block instead of using the same block + strcmp?